lib/gnat/services/service_responder.ex

```diff
@@ -0,0 +1,317 @@
+defmodule Gnat.Services.ServiceResponder do
+  @moduledoc false
+
+  require Logger
+  alias Gnat.Services.WireProtocol
+  @subscription_subject "$SRV.>"
+  @op_ping "PING"
+  @op_stats "STATS"
+  @op_info "INFO"
+
+  @idx_requests 1
+  @idx_errors 2
+  @idx_processing_time 3
+
+  @name_regex ~r/^[a-zA-Z0-9_-]+$/
+  @version_regex ~r/^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$/
+
+  use GenServer
+
+
+  @type state :: %{
+    config: Gnat.Services.Server.service_configuration(),
+    subject_map: map(),
+    connection_pid: pid,
+    instance_id: String.t,
+    subscription: non_neg_integer | String.t,
+    started: String.t
+  }
+
+  @spec start_link(map(), keyword()) :: GenServer.on_start
+  def start_link(settings, options \\ []) do
+    GenServer.start_link(__MODULE__, settings, options)
+  end
+
+  @impl GenServer
+  @spec init(map) :: {:ok, state()} | {:stop, String.t}
+  def init(settings) do
+    Process.flag(:trap_exit, true)
```

Comment:
I'm not seeing where we handle exit messages down below. Shouldn't we have a `handle_info` clause to accept exit messages? Or are we capturing exist just to get the `terminate` notification to try to unsubscribe?

Reply:
It's basically just to get the terminate. If it's better to handle the exit infos as well, I'll see if I can figure that out :)